wolfssl/ssl.h

```diff
@@ -2469,7 +2469,7 @@ struct DhKey;
 typedef int (*CallbackDhAgree)(WOLFSSL* ssl, struct DhKey* key,
         const unsigned char* priv, unsigned int privSz,
         const unsigned char* otherPubKeyDer, unsigned int otherPubKeySz,
-        unsigned char* out, unsigned int* outlen,
+        unsigned char* out, word32* outlen,
```

Comment:
Why does only outlen need to change and not the others?  I'd prefer not to change a public facing callback already in use.

Reply:
The integer value on a 16bit platform is 16bits, and the API is expecting 32bits.  So it needs to be changed in order to support 16bit platforms.

If this is a patch you guys do not want to merge, I can just have it separate, rebase it to get updates.